packages/actor-init-sparql/test/ActorInitSparql-test.ts

```diff
@@ -91,6 +92,8 @@ describe('ActorInitSparql', () => {
   describe('An ActorInitSparql instance', () => {
     const hypermedia = 'http://example.org/';
     const hypermedia2 = 'hypermedia@http://example.org/';
+    const hypermedia3 = 'http://username:passwd@example.org/';
+    const hypermedia4 = 'hypermedia@http://username:passwd@example.org/';
```

Comment:
Could you add some additional tests to check for more edge cases? Like empty username, empty password, encoded username, encoded password, ...

Might be easier to test this at `getSourceObjectFromString` directly.